raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -300,32 +300,54 @@ library NettingChannelLibrary {
         uint64 expiration;
         uint amount;
         bytes32 hashlock;
-        bytes32 h;
         bytes32 el;
+        bytes32 h;
 
         (expiration, amount, hashlock) = decodeLock(locked_encoded);
 
         if (self.locks[hashlock]) {
             throw;
         }
 
-        if (expiration < block.number)
+        if (expiration < block.number) {
             throw;
+        }
 
-        if (hashlock != sha3(secret))
+        if (hashlock != sha3(secret)) {
             throw;
+        }
         
         //Check if caller_address is a participant and select her partner
         uint8 index = 1 - index_or_throw(self, caller_address);
         
         Participant storage partner = self.participants[index];
+        if (partner.locksroot == 0) {
+            throw;
+        }
+
+        h = computeMerkleRoot(locked_encoded, merkle_proof);
+
+        if (partner.locksroot != h) {
+            throw;
+        }
+
+        partner.unlocked.push(Lock(expiration, amount, hashlock));
+        self.locks[hashlock] = true;
+    }
 
-        if (partner.nonce == 0) {
+    function computeMerkleRoot(bytes lock, bytes merkle_proof)
```

Comment:
I think `internal` may be a better choice for functions like these.
Reference: http://solidity.readthedocs.io/en/develop/contracts.html?highlight=internal#visibility-and-getters